community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveIntCollections.java

```diff
@@ -848,4 +840,33 @@ public static boolean contains( int[] values, int candidate )
             throw new IllegalStateException( "Encountered an already added item:" + item +
                     " when adding items uniquely to a collection:" + collection );
         }
-    }}
+    }
+
+    /**
+     * Deduplicates values in the {@code values} array.
+     *
+     * @param values sorted array of int values.
+     * @return the provided array if no duplicates were found, otherwise a new shorter array w/o duplicates.
```

Comment:
The implementation does not appear to require the input array to be sorted. And the test is also not using a sorted array. If it _did_ require the input array to be sorted, then this could be implemented in something akin to O(N) time. And if you don't need stable ordering, you could probably implement it by copying into a set in O(N log N) time. Not sure how much it matters, though, since N is probably often quite small here. You could do a sorted-ness check and, if sorted, a duplicate existence check in a single pass; each element must be strictly greater than the one before it.